src/pkg/retention/faked_controller.go

```diff
@@ -0,0 +1,88 @@
+package retention
+
+import (
+	"github.com/goharbor/harbor/src/pkg/retention/policy"
+	"github.com/goharbor/harbor/src/pkg/retention/q"
+)
+
+// FakedRetentionController ...
+type FakedRetentionController struct {
```

Comment:
can we define it into testing/pkg?

Reply:
> can we define it into testing/pkg?

It cannot be. Because the interface methods contains some struct in package `retention`, like `Execution` and `Task`